test/integration/dynamic-routing/test/middleware.test.js

```diff
@@ -0,0 +1,19 @@
+/* eslint-env jest */
+import './index.test'
+import fs from 'fs-extra'
+import { join } from 'path'
+
+const middlewarePath = join(__dirname, '../middleware.js')
+
+beforeAll(async () => {
+  await fs.writeFile(
+    middlewarePath,
+    `
+    import { NextResponse } from 'next/server'
+    export default function middleware() {
+      return NextResponse.next()
+    }
+  `
+  )
+})
+afterAll(() => fs.remove(middlewarePath))
```

Comment:
**question:** why does this test file contain no tests?

Reply:
This is re-running the same tests from `./index.test` but after adding a `middleware` file which is triggered just by importing the existing test file. 

Having the separate file that imports the tests allows them be run concurrently in CI. Can add a comment making this more clear.

Reply:
Ah!! I was note paying attention enough.
Please add a small comments for people like me!

Reply:
Yeah it was a bit hidden, updated in https://github.com/vercel/next.js/pull/37784/commits/1ea3288386ff0b1898e73084ccda64b64cac15ec